src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/AesCcm.OpenSsl.cs

```diff
@@ -18,7 +18,9 @@ private void ImportKey(ReadOnlySpan<byte> key)
         {
             // OpenSSL does not allow setting nonce length after setting the key
             // we need to store it as bytes instead
-            _key = key.ToArray();
+            // Pin the array on the POH so that the GC doesn't move it around to allow zeroing to be more effective.
+            _key = GC.AllocateArray<byte>(key.Length, pinned: true);
```

Comment:
KeyedHashAlgorithm.set_Key should probably change to this pattern, too.  And there might be other places that could also benefit from it.

Reply:
I'll follow up on this.